### PR TITLE
Browserstack opt in

### PR DESCRIPTION
**Pull request by Blackbaud-BobbyEarl**
The purpose of this PR is to read the `skyuxconfig.json` file (given to it via `skyPagesConfig` at https://github.com/blackbaud/skyux-builder/pull/431) and allow SPAs to define custom browsers to run against.

[Example 1](https://blackbaud.visualstudio.com/Products/_git/skyux-spa-browserstack-test/pullrequest/21746?_a=overview) shows the default now reverting back to a local Chrome browser.

[Example 2](https://blackbaud.visualstudio.com/Products/_git/skyux-spa-browserstack-test2/pullrequest/21757?_a=overview) shows a SPA with multiple `unit` and `e2e` browsers specified.

### CODE REVIEW

File: vsts/config/utils/browsers.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+module.exports = {
+  getBrowsers: (config, testSuite, defaults) => {
+
+    const browsers = get(config, `skyPagesConfig.skyux.testSettings.${testSuite}.browsers`, []);
+    const allowedPropertiesMap = map[testSuite];
+    const allowedPropertiesKeys = Object.keys(allowedPropertiesMap);
+
+    return browsers.map(browser => {
+
+      // Copies properties do we don't alter original.
```
Comment (Blackbaud-SteveBrush) [inline vsts/config/utils/browsers.js]: Should "do" be "so"?

**Comment by Blackbaud-SteveBrush** [comment]
@Blackbaud-BobbyEarl I'm seeing "undefined" in the results URL. Is that expected?
https://blackbaud.visualstudio.com/Products/Products%20Team/_build/results?buildId=527877&view=logs
screen shot 2018-07-05 at 10 06 17 am

**Comment by Blackbaud-BobbyEarl** [comment]
Good catch on the `undefined` message showing up @Blackbaud-SteveBrush.  That message shouldn't be showing up at all since the test suite isn't using Browserstack.  I'll update the PR shortly.

**Comment by Blackbaud-BobbyEarl** [comment]
Fixed the problem you found and fixed the linting errors.  Thanks @Blackbaud-SteveBrush.

**Pull request merged**